_core/admin/staff.php

```diff
@@ -0,0 +1,107 @@
+            error("This user already exists!");
+            
+        switch ($perm) {
+            case 'admin':
+                $allowed = 'janitor_board,moderator,admin';
+                $denied = 'none';
+                break;
+            case 'mod':
+                $allowed = 'janitor_board,moderator';
+                $denied = 'admin';
+                break;
+            case 'janitor':
+                $allowed = 'janitor_board';
+                $denied = 'moderator, admin';
+                break;
+            default:
+                error("Attempted to set unknown permission type.");
+                break;
+        }
+        
+        if ($pass1 !== $pass2)
+            error("Passwords did not match!");
+            
+        require_once(CORE_DIR . "/crypt/legacy.php");
+        $crypt = new SaguaroCryptLegacy;
+        $salt = $crypt->generate_hash($pass2);
+
+        mysql_query("INSERT INTO " . SQLMODSLOG . " (`user`, `password`, `public_salt`, `allowed`, `denied`) VALUES ('" . mysql_real_escape_string($user) . "', '" . $salt['hash'] . "', '" . $salt['public_salt'] . "', '" . $allowed . "', '" . $denied . "')");
+    }
+    
+    function remStaff($targUser = '', $actUser, $actPass) {
+        //remove staff member
+        $targUser = mysql_real_escape_string($targUser);
+        
+        if (!valid('admin'))
+            error("Permission denied");
+        if ($this->isStaff($targUser))
+            error("User doesn't exist! (GET error?");
+        if ($_COOKIE['saguaro_auser'] == $targUser)
+            error("You can't delete yourself!"); //oi ya cheeky shit ill bash yer fookin head in i sware on me mum
+        
+        mysql_query("DELETE FROM " . SQLMODSLOG . " WHERE user='" . $targUser ."'");
+    }
+    
+    function modifyStaff($targUser, $actUser, $actPass, $perms = array(), $self = 0) {
+        //modify existing user
+    }
+
+    function getStaff() {
+        //Staff list for panel
+        
+        $active = mysql_query(" SELECT * FROM " . SQLMODSLOG . "");
+        if ( !$result = mysql_call( "select * from " . SQLMODSLOG . "" ) )
+                echo S_SQLFAIL;
+        
+        if ( !$active ) 
+            echo S_SQLFAIL;
+        $j = 0;
+        $temp = '';
+        echo "<br><br>[<a href='" . PHP_ASELF_ABS . "'>Back to Panel</a><input type=hidden name=mode value=admin>]";
+        echo "<input type=hidden name=pass value=\"$pass\">\n";
+        echo "<div class=\"delbuttons\">";
+        $temp .= "<table class=\"postlists\"><br>";
+        $temp .=  "<tr class=\"postTable head\"><th>User</th><th>Allowed permissions</th><th>Denied permission</th><th>Modify user</th>";
+        $temp .=  "</tr>\n<form action='" . PHP_ASELF_ABS ."?mode=staff' method='get'>";
+
+        while ( $row = mysql_fetch_array( $active ) ) {
+                $j++;               
+                $class = ( $j % 2 ) ? "row1" : "row2"; //BG color
+                $temp .= "<tr class=$class>";
```

Comment:
My last statement about using CSS to determine alternating rows stands remains.

However, this still needs to be wrapped in `'`.